### PR TITLE
Cleanup ServerState.cpp

### PR DESCRIPTION
**Pull request by markuspf**
This PR cleans up some bits in `arangod/ServerState.cpp`

 * remove unused function `mkdir`
 * remove some out of date (and out of place) comments
 * move a function that tests for agency initialisation into a separate functions. This is mainly for readability.

This change is a trivial rework / code cleanup without additional test coverage,
this change is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

**Comment by markuspf** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/6079/

**Comment by markuspf** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/6080/

**Comment by markuspf** [comment]
Tests are blue.

File: arangod/Agency/AgencyStrings.h [hunk 1]
```diff
@@ -21,6 +21,9 @@
 /// @author Kaveh Vahedipour
 ////////////////////////////////////////////////////////////////////////////////
 
+#if !defined(ARANGOD_AGENCY_AGENCY_STRINGS_H)
```
Comment (goedderz) [inline arangod/Agency/AgencyStrings.h]:
```suggestion
#ifndef ARANGOD_AGENCY_AGENCY_STRINGS_H
```

File: arangod/Cluster/ServerState.cpp [hunk 2]
```diff
@@ -626,8 +596,26 @@ bool ServerState::registerAtAgencyPhase1(AgencyComm& comm, ServerState::RoleEnum
...
+bool ServerState::registerAtAgencyPhase1(AgencyComm& comm, ServerState::RoleEnum const& role) {
+  std::string const agencyListKey = roleToAgencyListKey(role);
+  std::string const latestIdKey = "Latest" + roleToAgencyKey(role) + "Id";
+
+  VPackBuilder builder;
+  builder.add(VPackValue("none"));
+
+  if(!checkIfAgencyInitialized(comm, role)) {
     return false;
   }
```
Comment (goedderz) [inline arangod/Cluster/ServerState.cpp]: You could create the builder after the check, would make the early-exit a little easier to spot.

**Comment by markuspf** [comment]
Addressed @goedderz 2 comments, and started another jenkins to be sure:  https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6134/

**Pull request merged**